src/subscript/welltest_dpds/welltest_dpds.py

```diff
@@ -0,0 +1,472 @@
+  - dpds_lag2 eg superpositioned time derivative of pressure lag 2
+  - sspt superpositioned time
+  - wbhp vs cum time
+  - wwpr vs cum time
+  - wopr vs cum time if --phase == OIL
+  - wgpr vs cum time if --phase == GAS
+
+"""
+
+"""
+
+Script is a rewrite of a legacy script originally developed and improved by:
+ * Jon Sætrom
+ * Bjørn Kåre Hegstad
+ * Cecile Otterlei
+ * Hodjat Moradi
+
+AUTHOR: Eivind Smørgrav
+
+TODO
+ - Support pseudo pressure vs time relevant for gas and gas condensate fields.
+ - Make error handling more robust, eg:
+     - check if all necessary vectors are present wbhp, wopr or wgpr, wwpr is optional
+- decide how the output files should be handled, there are many and messy as of now
+     - suggestion unify all the ones with two columns into one csv file
+     - arg options to define which vectors to output
+     - lift the wwpr requirement, not used, but legacy script  outputted it
+"""
+
+CATEGORY = "modelling.reservoir"
+
+EXAMPLES = """
+.. code-block:: console
+
+ FORWARD_MODEL WELLTEST_DPDS(<ECLBASE>, <WELLNAME>=DST_WELL)
+
+ FORWARD_MODEL  WELLTEST_DPDS(<ECLBASE>, <WELLNAME>=OP_1, <PHASE>=GAS,
+    <BUILDUP_NR>=1, <OUTPUTDIRECTORY>=dst, <OUTFILESSUFIX>=OP_1_1)
+
+"""
+
+
+def get_parser():
+    """
+    Define the argparse parser
+
+    Returns:
+        parser (argparse.ArgumentParser)
+    """
+    parser = argparse.ArgumentParser(
+        description=DESCRIPTION,
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "eclcase",
+        type=str,
+        help="Eclipse case to extract from",
+    )
+    parser.add_argument(
+        "wellname",
+        type=str,
+        help="Name of well to extract results from",
+    )
+    parser.add_argument(
+        "--outfilessuffix",
+        type=str,
+        help='Suffix to be added to result files. Default: ""',
+        default="",
+    )
+    parser.add_argument(
+        "-n",
+        "--buildup_nr",
+        type=int,
+        help="Buildup nr, eg which buildup to extract. Counting from 1. Default: 1 ",
+        default=1,
+    )
+    parser.add_argument(
+        "-o",
+        "--outputdirectory",
+        type=str,
+        help="Directory to put the output files. Detault: .",
+        default="./",
+    )
+    parser.add_argument(
+        "--phase",
+        type=str,
+        choices=["OIL", "GAS"],
+        help="Main fluid phase in test (OIL/GAS). Default: OIL",
+        default="OIL",
+        required=False,
+    )
+    return parser
+
+
+def get_summary_vec(summary, key):
+    """
+    Read vector corresponding to key from summary instance
+
+    Args:
+       summary: (ecl.summary.EclSum)
+       key: (str)
+
+    Returns:
+        vec  : np.array
+    """
+
+    try:
+        return summary.numpy_vector(key)
+    except KeyError:
+        print("No such key in summary file:" + key)
+        raise
+
+
+def get_buildup_indices(rates):
```

Comment:
The `get_` prefix to many of this functions can probably just be dropped. These are not typical "getters"